File: ,/chinese/Windows/src/05-Singleton/Q3/Singleton.java
public class Singleton {
    private static Singleton singleton = null;
    private Singleton() {
        System.out.println("已產生物件個體。");
    }
    public static Singleton getInstance() {
        if (singleton == null) {
            singleton = new Singleton();
        }
        return singleton;
    }
}

File: 品味Java的21種設計模式/dp_cpp/singleton/example3/Singleton.cpp
#include "Singleton.h"

namespace cn {
namespace javass {
namespace dp {
namespace singleton {
namespace example3 {

Singleton::Singleton() {
    //
}

Singleton Singleton::getInstance() {
    //直接使用已经创建好的实例
    return uniqueInstance;
}

void Singleton::singletonOperation() {
    //功能处理
}

std::string Singleton::getSingletonData() {
    return singletonData;
}
}
}
}
}
}
